packages/remix-react/transition.ts

```diff
@@ -596,6 +600,28 @@ export function createTransitionManager(init: TransitionManagerInit) {
     }
   }
 
+  function isIndexRequestUrl(url: URL) {
+    let indexRequest = false;
+
+    for (let param of url.searchParams.getAll("index")) {
+      if (!param) {
+        indexRequest = true;
+      }
+    }
+
+    return indexRequest;
```

Comment:
```suggestion
  function isIndexRequestUrl(url: URL) {
    for (let param of url.searchParams.getAll("index")) {
      // only use bare `?index` params without a value
      // ✅ /foo?index
      // ✅ /foo?index&index=123
      // ✅ /foo?index=123&index
      // ❌ /foo?index=123
      if (param === "") return true;
    }
    
    return false;
```

Reply:
Seems worth calling out for "future us".

Reply:
Good call.